patches/api/0384-Event-Effect-API-Improvements.patch

```diff
@@ -0,0 +1,775 @@
+new file mode 100644
+index 0000000000000000000000000000000000000000..a75bf919b8606710d3eda93c3fc8e373a12af0ab
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/block/event/SimpleBlockEvent.java
+@@ -0,0 +1,12 @@
++package io.papermc.paper.block.event;
++
++import org.bukkit.Material;
++
++import java.util.Set;
++
++public class SimpleBlockEvent extends BlockEvent {
++
++    SimpleBlockEvent(Set<Material> materials, int id) {
++        super(materials, id);
++    }
++}
+diff --git a/src/main/java/io/papermc/paper/world/event/ConfigurableWorldEvent.java b/src/main/java/io/papermc/paper/world/event/ConfigurableWorldEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..f3a0bde72d5e8b9edca46ea2f376359f8ae4b898
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/world/event/ConfigurableWorldEvent.java
+@@ -0,0 +1,18 @@
++package io.papermc.paper.world.event;
++
++import org.jetbrains.annotations.ApiStatus;
++
++public class ConfigurableWorldEvent<T> extends WorldEvent {
++
++    private final Class<T> clazz;
++
++    ConfigurableWorldEvent(int id, boolean isGlobal, Class<T> clazz) {
++        super(id, isGlobal);
++        this.clazz = clazz;
++    }
++
++    @ApiStatus.Internal
++    public Class<T> getConfigurationClass() {
++        return clazz;
++    }
++}
+diff --git a/src/main/java/io/papermc/paper/world/event/SculkChargeEventConfiguration.java b/src/main/java/io/papermc/paper/world/event/SculkChargeEventConfiguration.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..59f77614f8c2e21c07be44be00f2a8b4e3b450c9
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/world/event/SculkChargeEventConfiguration.java
+@@ -0,0 +1,26 @@
++package io.papermc.paper.world.event;
++
++import org.bukkit.block.BlockFace;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Range;
++
++import java.util.Collection;
++
++public record SculkChargeEventConfiguration(@Range(from = 0, to = 1000) int charge, @NotNull Collection<BlockFace> blockFaces) {
++
++    /**
++     * Gets the charge of this current sculk charge event.
++     */
++    @Override
++    public int charge() {
++        return this.charge;
++    }
++
++    /**
++     * Gets the block faces that this sculk charge effects.
++     */
++    @NotNull
++    public Collection<BlockFace> blockFaces() {
++        return this.blockFaces;
++    }
++}
+diff --git a/src/main/java/io/papermc/paper/world/event/SimpleWorldEvent.java b/src/main/java/io/papermc/paper/world/event/SimpleWorldEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..543b040e93b3e1fe875ac79508ffb8fe69ddeb95
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/world/event/SimpleWorldEvent.java
+@@ -0,0 +1,8 @@
++package io.papermc.paper.world.event;
++
++public class SimpleWorldEvent extends WorldEvent {
```

Comment:
Should be a non-public innerclass of WorldEvent. The less public types we expose to the API, the better. Although, what is the point of this? Why not make WorldEvent not-abstract and still extend it for the other types?